compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -70,6 +70,210 @@ void compute_asym_scale_zp(float min, float max, float &scaling_factor, int64_t
   zp = nudged_zero_point;
 }
 
+bool get_channel_dim_index(CircleConst *node, int *len_per_dim, int &channel_dim_index)
+{
+  auto succs = loco::succs(node);
+  if (succs.size() != 1) // assume weights is used by only one node
+    return false;
+
+  for (auto out : succs)
+  {
+    auto conv = dynamic_cast<CircleConv2D *>(out);
+    auto dw_conv = dynamic_cast<CircleDepthwiseConv2D *>(out);
+    auto tw_conv = dynamic_cast<CircleTransposeConv *>(out);
+    auto fc = dynamic_cast<CircleFullyConnected *>(out);
+
+    if (conv != nullptr && conv->filter() == node)
+    {
+      len_per_dim[0] = node->dim(0).value();
+      len_per_dim[1] = node->dim(1).value();
+      len_per_dim[2] = node->dim(2).value();
+      len_per_dim[3] = node->dim(3).value();
+      channel_dim_index = 3;
+      return true;
+    }
+    else if ((tw_conv != nullptr && tw_conv->filter() == node) ||
+             (dw_conv != nullptr && dw_conv->filter() == node))
+    {
+      len_per_dim[0] = node->dim(0).value();
+      len_per_dim[1] = node->dim(1).value();
+      len_per_dim[2] = node->dim(2).value();
+      len_per_dim[3] = node->dim(3).value();
+      channel_dim_index = 2;
+      return true;
+    }
+    else if (fc != nullptr && fc->weights() == node)
+    {
+      len_per_dim[0] = 1;
+      len_per_dim[1] = 1;
+      len_per_dim[2] = node->dim(0).value();
+      len_per_dim[3] = node->dim(1).value();
+      channel_dim_index = 3;
+      return true;
+    }
+    else
+    {
+      // node does not support channle-wise quantization
+      assert(false);
+    }
+  }
+
+  return false;
+}
+
+int cal_offset(int *len_per_dim, int *indices)
+{
+  return indices[0] * len_per_dim[1] * len_per_dim[2] * len_per_dim[3] +
+         indices[1] * len_per_dim[2] * len_per_dim[3] + indices[2] * len_per_dim[3] + indices[3];
+}
+
+void cal_minmax_per_channel(CircleConst *node, std::vector<float> &min, std::vector<float> &max)
+{
+  const int len = 4;
+  static_assert(len == 4, "all weights should be set to the same rank as the convolution weights");
+  int len_per_dim[len] = {
+      0,
+  };
+  int indices[len] = {
+      0,
+  };
+  int channel_dim_index;
+  int size;
```

Comment:
This suggestion is for static analysis warning issue
```suggestion
  int channel_dim_index{0};
  int size{0};
```